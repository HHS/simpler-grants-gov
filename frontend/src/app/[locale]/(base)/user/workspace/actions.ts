"use server";

import { getSession } from "src/services/auth/session";
import { inviteUserToOrganization } from "src/services/fetch/fetchers/organizationsFetcher";
import { OrganizationInviteRecord } from "src/types/organizationTypes";
import { z } from "zod";

import { getTranslations } from "next-intl/server";

export type OrganizationInviteValidationErrors = {
  email?: string[];
  role?: string[];
};

export type OrganizationInviteResponse = {
  data?: OrganizationInviteRecord;
  errorMessage?: string;
  validationErrors?: OrganizationInviteValidationErrors;
  success?: boolean;
};

const validateInviteUserAction = async (formData: FormData) => {
  const t = await getTranslations("ManageUsers.inviteUser.validationErrors");
  const schema = z.object({
    email: z.string().min(1, { message: t("email") }),
    role: z.string().min(1, { message: t("role") }),
  });

  const validatedFields = schema.safeParse({
    email: formData.get("email"),
    role: formData.get("role"),
  });

  // Return early if the form data is invalid (server side validation!)
  if (!validatedFields.success) {
    return validatedFields.error.flatten().fieldErrors;
  }
};

export const inviteUserAction = async (
  _prevState: unknown,
  formData: FormData,
  organizationId: string,
): Promise<OrganizationInviteResponse> => {
  const session = await getSession();

  if (!session || !session.token || !session.user_id) {
    return {
      errorMessage: "Not logged in",
    };
  }

  const validationErrors = await validateInviteUserAction(formData);
  if (validationErrors) {
    return {
      validationErrors,
    };
  }

  const requestData = {
    email: formData.get("email") as string,
    roleId: formData.get("role") as string,
    organizationId,
  };

  let inviteUserResponse: OrganizationInviteRecord;
  try {
    inviteUserResponse = await inviteUserToOrganization(
      session.token,
      requestData,
    );
    return { data: inviteUserResponse, success: true };
  } catch (e) {
    const error = e as Error;
    console.error(
      `Error inviting user to org - ${error.message} ${error.cause?.toString() || ""}`,
    );
    return {
      errorMessage: error.message,
    };
  }
};
