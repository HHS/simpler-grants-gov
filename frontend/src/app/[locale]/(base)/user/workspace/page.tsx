import { Metadata } from "next";
import { getSession } from "src/services/auth/session";
import withFeatureFlag from "src/services/featureFlags/withFeatureFlag";
import { getUserOrganizations } from "src/services/fetch/fetchers/organizationsFetcher";
import { getUserDetails } from "src/services/fetch/fetchers/userFetcher";
import { Organization } from "src/types/applicationResponseTypes";
import { LocalizedPageProps } from "src/types/intl";

import { useTranslations } from "next-intl";
import { getTranslations } from "next-intl/server";
import Link from "next/link";
import { redirect } from "next/navigation";
import { ErrorMessage, GridContainer } from "@trussworks/react-uswds";

export async function generateMetadata({
  params,
}: LocalizedPageProps): Promise<Metadata> {
  const { locale } = await params;
  const t = await getTranslations({ locale });
  const meta: Metadata = {
    title: t("UserWorkspace.pageTitle"),
    description: t("Index.metaDescription"),
  };
  return meta;
}

const UserOrganizationsList = ({
  userOrganizations,
}: {
  userOrganizations: Organization[];
}) => {
  const t = useTranslations("UserWorkspace");
  return (
    <>
      <h2>{t("organizations")}</h2>
      <ul>
        {userOrganizations.map((userOrganization) => (
          <li key={userOrganization.organization_id}>
            <Link href={`/organization/${userOrganization.organization_id}`}>
              {userOrganization.sam_gov_entity.legal_business_name}
            </Link>
          </li>
        ))}
      </ul>
    </>
  );
};

async function UserWorkspace() {
  const t = await getTranslations("UserWorkspace");

  const session = await getSession();
  if (!session?.email) {
    // this won't happen, as email is required on sessions, and we're wrapping this in an auth gate in the layout
    console.error("no user session, or user has no email address");
    return;
  }
  let userDetails;
  let userOrganizations;
  const userDetailsPromise = getUserDetails(session.token, session.user_id);
  const userOrganizationsPromise = getUserOrganizations(
    session.token,
    session.user_id,
  );
  try {
    [userDetails, userOrganizations] = await Promise.all([
      userDetailsPromise,
      userOrganizationsPromise,
    ]);
  } catch (e) {
    console.error("Unable to fetch user details", e);
  }

  return (
    <GridContainer className="padding-top-2 tablet:padding-y-6">
      <h1>{t("title")}</h1>
      {userDetails && userOrganizations ? (
        <UserOrganizationsList userOrganizations={userOrganizations} />
      ) : (
        <ErrorMessage>{t("fetchError")}</ErrorMessage>
      )}
    </GridContainer>
  );
}

export default withFeatureFlag<object, never>(
  UserWorkspace,
  "userAdminOff",
  () => redirect("/maintenance"),
);
