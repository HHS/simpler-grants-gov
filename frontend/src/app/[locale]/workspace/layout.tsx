import { LayoutProps } from "src/types/generalTypes";

import BetaAlert from "src/components/BetaAlert";
import { AuthenticationGate } from "src/components/user/AuthenticationGate";

export default function WorkspaceLayout({ children }: LayoutProps) {
  return (
    <AuthenticationGate>
      <BetaAlert containerClasses="margin-bottom-5" />
      {children}
    </AuthenticationGate>
  );
}
