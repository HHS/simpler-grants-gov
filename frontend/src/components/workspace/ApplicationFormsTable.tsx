import { ApplicationFormDetail } from "src/types/applicationResponseTypes";
import { CompetitionForms } from "src/types/competitionsResponseTypes";
import { FormDetail } from "src/types/formResponseTypes";

import { useTranslations } from "next-intl";
import Link from "next/link";
import { Table } from "@trussworks/react-uswds";

import { USWDSIcon } from "src/components/USWDSIcon";

export const selectFormsByRequired = ({
  forms,
  required,
}: {
  forms: CompetitionForms;
  required: boolean;
}): FormDetail[] => {
  return forms.reduce<FormDetail[]>((acc, item) => {
    if (item.is_required === required) {
      acc.push(item.form);
    }
    return acc;
  }, []);
};

const selectApplicationForm = ({
  applicationForms,
  formId,
}: {
  applicationForms: ApplicationFormDetail[];
  formId: string;
}) => {
  return applicationForms
    ? applicationForms.find(
        (applicationForm) => applicationForm.form_id === formId,
      )
    : null;
};

export const ApplicationFormsTable = ({
  applicationForms,
  applicationId,
  forms,
}: {
  applicationForms: ApplicationFormDetail[];
  applicationId: string;
  forms: CompetitionForms;
}) => {
  const requiredForms = selectFormsByRequired({ forms, required: true });
  const conditionalRequiredForms = selectFormsByRequired({
    forms,
    required: false,
  });
  const t = useTranslations("Application.competitionFormTable");

  return (
    <>
      <h3>{t("requiredForms")}</h3>
      <ApplicationTable
        forms={requiredForms}
        applicationForms={applicationForms}
        applicationId={applicationId}
      />
      {conditionalRequiredForms.length > 0 && (
        <>
          <h3>{t("conditionalForms")}</h3>
          <p>{t("conditionalFormsDescription")}</p>
          <ApplicationTable
            forms={conditionalRequiredForms}
            applicationForms={applicationForms}
            applicationId={applicationId}
          />
        </>
      )}
    </>
  );
};

const selectApplicationFormById = ({
  forms,
  formId,
}: {
  forms: FormDetail[];
  formId: string;
}): FormDetail | undefined => {
  return forms.find((form) => form.form_id === formId);
};

const ApplicationTable = ({
  applicationForms,
  applicationId,
  forms,
}: {
  applicationForms: ApplicationFormDetail[];
  applicationId: string;
  forms: FormDetail[];
}) => {
  const t = useTranslations("Application.competitionFormTable");

  return (
    <Table className="width-full overflow-wrap">
      <thead>
        <tr>
          <th scope="col" className="bg-base-lightest padding-y-205">
            {t("status")}
          </th>
          <th scope="col" className="bg-base-lightest padding-y-205">
            {t("form")}
          </th>
          <th scope="col" className="bg-base-lightest padding-y-205">
            {t("instructions")}
          </th>
          <th scope="col" className="bg-base-lightest padding-y-205">
            {t("updated")}
          </th>
        </tr>
      </thead>
      <tbody>
        {applicationForms.map((form, index) => (
          <tr key={index}>
            <td data-label={t("status")}>
              <CompetitionStatus
                applicationForms={applicationForms}
                formId={form.form_id}
              />
            </td>
            <td data-label={t("form")}>
              <FormLink
                formId={form.form_id}
                forms={forms}
                applicationId={applicationId}
                appFormId={form.application_form_id}
              />
            </td>
            <td data-label={t("instructions")}>
              <InstructionsLink
                forms={forms}
                formId={form.form_id}
                text={t("downloadInstructions")}
                unavailableText={t("attachmentUnavailable")}
              />
            </td>
            <td data-label={t("updated")}>
              <div> -- </div>
            </td>
          </tr>
        ))}
      </tbody>
    </Table>
  );
};

const CompetitionStatus = ({
  formId,
  applicationForms,
}: {
  applicationForms: ApplicationFormDetail[];
  formId: string;
}) => {
  const t = useTranslations("Application.competitionFormTable.statuses");
  const applicationForm = selectApplicationForm({ formId, applicationForms });

  if (applicationForm?.application_form_status === "in_progress") {
    return (
      <div className="display-flex flex-align-center text-bold icon-active">
        <USWDSIcon name="loop" className="margin-right-2px" />
        {t("in_progress")}
      </div>
    );
  } else if (applicationForm?.application_form_status === "complete") {
    return (
      <div className="display-flex flex-align-center text-bold">
        <USWDSIcon name="check" className="text-primary margin-right-2px" />
        {t("complete")}
      </div>
    );
  } else {
    return <>-</>;
  }
};

const InstructionsLink = ({
  formId,
  forms,
  text,
  unavailableText,
}: {
  formId: string;
  forms: FormDetail[];
  text: string;
  unavailableText: string;
}) => {
  const instructions = selectApplicationFormById({
    forms,
    formId,
  })?.form_instruction;
  const downloadPath = instructions?.download_path || null;
  return (
    <>
      {downloadPath ? (
        <Link
          className="display-flex flex-align-center font-sans-2xs"
          href={downloadPath}
        >
          <USWDSIcon name="save_alt" className="margin-right-05" />
          {text}
        </Link>
      ) : (
        <>{unavailableText}</>
      )}
    </>
  );
};

const FormLink = ({
  formId,
  forms,
  applicationId,
  appFormId,
}: {
  formId: string;
  forms: FormDetail[];
  applicationId: string;
  appFormId: string;
}) => {
  const formName = selectApplicationFormById({
    forms,
    formId,
  })?.form_name;

  return (
    <>
      {formName && (
        <Link
          className="text-bold"
          href={`/workspace/applications/application/${applicationId}/form/${appFormId}`}
        >
          {formName}
        </Link>
      )}
    </>
  );
};
