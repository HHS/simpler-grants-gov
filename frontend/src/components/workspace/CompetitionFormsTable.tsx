import { CompetitionForms } from "src/types/competitionsResponseTypes";
import { FormDetail, FormInstruction } from "src/types/formResponseTypes";

import { useTranslations } from "next-intl";
import Link from "next/link";
import { Table } from "@trussworks/react-uswds";

import { USWDSIcon } from "src/components/USWDSIcon";

export const CompetitionFormsTable = ({
  forms,
  applicationId,
}: {
  forms: CompetitionForms;
  applicationId: string;
}) => {
  const requiredForms = selectForms({ forms, required: true });
  const conditionalRequiredForms = selectForms({ forms, required: false });
  const t = useTranslations("Application.competitionFormTable");

  return (
    <>
      <h3>{t("requiredForms")}</h3>

      <CompetitionTable forms={requiredForms} applicationId={applicationId} />
      <h3>{t("conditionalForms")}</h3>

      <CompetitionTable
        forms={conditionalRequiredForms}
        applicationId={applicationId}
      />
    </>
  );
};

export const selectForms = ({
  forms,
  required,
}: {
  forms: CompetitionForms;
  required: boolean;
}): FormDetail[] => {
  return forms.reduce<FormDetail[]>((acc, item) => {
    if (item.is_required === required) {
      acc.push(item.form);
    }
    return acc;
  }, []);
};

const CompetitionTable = ({
  forms,
  applicationId,
}: {
  forms: FormDetail[];
  applicationId: string;
}) => {
  const t = useTranslations("Application.competitionFormTable");

  return (
    <Table className="width-full overflow-wrap">
      <thead>
        <tr>
          <th scope="col" className="bg-base-lighter">
            {t("status")}
          </th>
          <th scope="col" className="bg-base-lighter">
            {t("form")}
          </th>
          <th scope="col" className="bg-base-lighter">
            {t("attachment")}
          </th>
          <th scope="col" className="bg-base-lighter">
            {t("instructions")}
          </th>
          <th scope="col" className="bg-base-lighter">
            {t("updated")}
          </th>
        </tr>
      </thead>
      <CompetitionTableBody forms={forms} applicationId={applicationId} />
    </Table>
  );
};

const CompetitionTableBody = ({
  forms,
  applicationId,
}: {
  forms: FormDetail[];
  applicationId: string;
}) => {
  const t = useTranslations("Application.competitionFormTable");

  return (
    <tbody>
      {forms.map((form, index) => (
        <tr key={index}>
          <td data-label={t("status")}>--</td>
          <td data-label={t("form")}>
            <Link
              className="text-bold"
              href={`/workspace/applications/application/${applicationId}/form/${form.form_id}`}
            >
              {form.form_name}
            </Link>
          </td>
          <td data-label={t("attachment")}>
            <div> -- </div>
          </td>
          <td data-label={t("instructions")}>
            <InstructionsLink
              instsruction={form.form_instruction}
              text={t("downloadInstructions")}
            />
          </td>
          <td data-label={t("updated")}>
            <div> -- </div>
          </td>
        </tr>
      ))}
    </tbody>
  );
};

const InstructionsLink = ({
  instsruction,
  text,
}: {
  instsruction: FormInstruction;
  text: string;
}) => {
  return (
    <Link
      className="display-flex flex-align-center font-sans-2xs"
      href={instsruction.download_path}
    >
      <USWDSIcon name="save_alt" className="margin-right-05" />
      {text}
    </Link>
  );
};
