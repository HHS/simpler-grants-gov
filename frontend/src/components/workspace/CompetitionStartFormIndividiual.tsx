"use client";

import { useTranslations } from "next-intl";
import { RefObject } from "react";
import {
  Button,
  ErrorMessage,
  FormGroup,
  Label,
  ModalFooter,
  ModalRef,
  ModalToggleButton,
  TextInput,
} from "@trussworks/react-uswds";

const CompetitionStartFormIndividiual = ({
  opportunityTitle,
  error = "",
  loading = false,
  modalRef,
  onChange,
  onClose,
  onSubmit,
  validationError = "",
}: {
  opportunityTitle: string;
  loading?: boolean;
  error?: string;
  modalRef: RefObject<ModalRef | null>;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  onClose: () => void;
  onSubmit: () => void;
  validationError?: string;
}) => {
  const t = useTranslations("OpportunityListing");

  return (
    <div className="display-flex flex-align-start">
      <FormGroup error={!!validationError} className="margin-top-1">
        <p className="font-sans-sm" data-testid="opportunity-title">
          {opportunityTitle}
        </p>
        <p className="font-sans-3xs">
          {t("startApplicationModal.requiredText")}
        </p>
        <Label
          id={`label-for-name`}
          key={`label-for-name`}
          htmlFor="application-name"
          className="font-sans-2xs"
        >
          {t("startApplicationModal.name")}
          <span>
            <br /> {t("startApplicationModal.description")}
          </span>
        </Label>
        {validationError && <ErrorMessage>{validationError}</ErrorMessage>}
        {error && <ErrorMessage>{error}</ErrorMessage>}

        <TextInput
          type="text"
          name="application-name"
          id="application-name"
          onChange={onChange}
        />
        <ModalFooter>
          <Button
            onClick={onSubmit}
            type="button"
            data-testid="competition-start-individual-save"
            disabled={!!loading}
          >
            {loading ? "loading..." : t("startApplicationModal.saveButtonText")}
          </Button>
          <ModalToggleButton
            modalRef={modalRef}
            closer
            unstyled
            className="padding-105 text-center"
            onClick={onClose}
          >
            {t("startApplicationModal.cancelButtonText")}
          </ModalToggleButton>
        </ModalFooter>
      </FormGroup>
    </div>
  );
};

export default CompetitionStartFormIndividiual;
