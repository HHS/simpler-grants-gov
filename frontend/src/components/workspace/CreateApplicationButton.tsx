"use client";

import { useUser } from "src/services/auth/useUser";
import { startApplication } from "src/services/fetch/fetchers/clientApplicationFetcher";

import { useRouter } from "next/navigation";
import { useCallback, useState } from "react";
import { Button } from "@trussworks/react-uswds";

const CreateApplicationButton = () => {
  const router = useRouter();
  const { user } = useUser();
  const disabled = !user?.token;

  const [loading, setLoading] = useState<boolean>();

  const handleSubmit = useCallback(() => {
    if (!user?.token) return;
    setLoading(true);
    startApplication(user.token)
      .then((data) => {
        const { applicationId } = data;
        router.push(`/formPrototype/${applicationId}`);
      })
      .catch((error) => {
        console.error(error);
      })
      .finally(() => {
        setLoading(false);
      });
  }, [router, user]);
  return (
    <Button disabled={disabled} onClick={handleSubmit} type="submit">
      {loading ? "loading..." : "Create new application"}
    </Button>
  );
};

export default CreateApplicationButton;
