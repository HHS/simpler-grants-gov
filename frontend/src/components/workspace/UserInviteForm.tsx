"use client";

import { inviteUserAction } from "src/app/[locale]/(base)/user/workspace/actions";
import { UserRole } from "src/types/userTypes";

import { useTranslations } from "next-intl";
import { useActionState, useMemo } from "react";
import {
  Button,
  FormGroup,
  Label,
  Select,
  TextInput,
} from "@trussworks/react-uswds";

export const inviteUserActionForOrganization =
  (organizationId: string) => (_prevState: unknown, formData: FormData) =>
    inviteUserAction(_prevState, formData, organizationId);

export const rolesToOptions = (roles: UserRole[]) => {
  return roles.map((role) => (
    <option key={role.role_id} value={role.role_id}>
      {role.role_name}
    </option>
  ));
};
export function UserInviteButton({ success = false, disabled = false }) {
  const t = useTranslations("ManageUsers.inviteUser.button");
  if (success) {
    return <div>{t("success")}</div>;
  }
  return (
    <Button disabled={disabled} type="submit">
      {t("label")}
    </Button>
  );
}

export function UserInviteForm({
  organizationId,
  roles,
}: {
  organizationId: string;
  roles: UserRole[];
}) {
  // handle form action and state (server action)
  const t = useTranslations("ManageUsers.inviteUser");

  const inviteUser = useMemo(
    () => inviteUserActionForOrganization(organizationId),
    [organizationId],
  );
  const roleOptions = useMemo(() => rolesToOptions(roles), [roles]);

  const [state, formAction, isPending] = useActionState(inviteUser, {
    success: false,
  });

  return (
    <form action={formAction}>
      <FormGroup>
        <Label htmlFor="email">{t("inputs.email.label")}</Label>
        <TextInput
          name="email"
          id="inviteUser-email"
          type="email"
          placeholder={t("inputs.email.placeholder")}
          value={state.data?.invitee_email}
          disabled={isPending}
        />
        <Label htmlFor="email">{t("inputs.role.label")}</Label>
        <Select
          name="role"
          id="inviteUser-role"
          defaultValue={t("inputs.role.placeholder")}
          disabled={isPending}
          value={state.data?.roles[0].role_name}
        >
          {roleOptions}
        </Select>
      </FormGroup>
      <UserInviteButton />
    </form>
  );
}
