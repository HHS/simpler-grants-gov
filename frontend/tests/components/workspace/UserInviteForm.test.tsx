import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { noop } from "lodash";
import { fakeUserRole } from "src/utils/testing/fixtures";
import { useTranslationsMock } from "src/utils/testing/intlMocks";

import { UserInviteForm } from "src/components/workspace/UserInviteForm";

const mockUseActionState = jest.fn();

jest.mock("react", () => ({
  ...jest.requireActual<typeof import("react")>("react"),
  useActionState: () => mockUseActionState() as unknown,
}));

jest.mock("src/app/[locale]/(base)/user/workspace/actions", () => ({
  inviteUserAction: noop,
}));

jest.mock("next-intl", () => ({
  useTranslations: () => useTranslationsMock(),
}));

describe("UserInviteForm", () => {
  afterEach(() => {
    jest.resetAllMocks();
  });
  it("matches snapshot", () => {
    mockUseActionState.mockReturnValue([{}, noop, false]);
    const { container } = render(
      <UserInviteForm organizationId="1" roles={[fakeUserRole]} />,
    );
    expect(container).toMatchSnapshot();
  });

  it("renders validation errors", () => {
    mockUseActionState.mockReturnValue([
      { validationErrors: { email: ["some error"] } },
      noop,
      false,
    ]);

    render(<UserInviteForm organizationId="1" roles={[fakeUserRole]} />);

    const alert = screen.getByRole("alert");

    expect(alert).toBeInTheDocument();
    expect(alert).toHaveTextContent("some error");
  });
  it("displays error message on error", () => {
    mockUseActionState.mockReturnValue([
      {
        errorMessage: "big error",
      },
      noop,
      false,
    ]);

    render(<UserInviteForm organizationId="1" roles={[fakeUserRole]} />);

    const alert = screen.getByRole("heading");
    expect(alert).toBeInTheDocument();
    expect(alert).toHaveTextContent("errorHeading");
  });
  it("calls action with expected values on submit", async () => {
    const mockAction = jest.fn();
    const expectedFormData = new FormData();
    expectedFormData.append("role", fakeUserRole.role_id);
    expectedFormData.append("email", "an@email.address");

    mockUseActionState.mockReturnValue([
      {
        success: false,
      },
      mockAction,
      false,
    ]);

    render(<UserInviteForm organizationId="1" roles={[fakeUserRole]} />);

    const textInput = screen.getByRole("textbox");
    const select = screen.getByRole("combobox");
    const submitButton = screen.getByRole("button");

    await userEvent.type(textInput, "an@email.address");
    await userEvent.selectOptions(select, fakeUserRole.role_name);
    await userEvent.click(submitButton);

    expect(mockAction).toHaveBeenCalled();
  });
});
