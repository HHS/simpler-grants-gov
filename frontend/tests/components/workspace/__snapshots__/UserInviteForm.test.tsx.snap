// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInviteForm matches snapshot 1`] = `
<div>
  <form
    action="javascript:throw new Error('A React form was unexpectedly submitted. If you called form.submit() manually, consider using form.requestSubmit() instead. If you\\'re trying to use event.stopPropagation() in a submit event handler, consider also calling event.preventDefault().')"
  >
    <div
      class="grid-row grid-gap-md"
      data-testid="grid"
    >
      <div
        class="tablet:grid-col-4"
        data-testid="grid"
      >
        <div
          class="usa-form-group"
          data-testid="formGroup"
        >
          <label
            class="usa-label"
            data-testid="label"
            for="email"
          >
            inputs.email.label
            <span
              class="usa-hint usa-hint--required"
              data-testid="required-field-indicator"
            >
               *
            </span>
          </label>
          <input
            class="usa-input"
            data-testid="textInput"
            id="inviteUser-email"
            name="email"
            placeholder="inputs.email.placeholder"
            type="email"
          />
        </div>
      </div>
      <div
        class="tablet:grid-col-4"
        data-testid="grid"
      >
        <div
          class="usa-form-group"
          data-testid="formGroup"
        >
          <label
            class="usa-label"
            data-testid="label"
            for="email"
          >
            inputs.role.label
            <span
              class="usa-hint usa-hint--required"
              data-testid="required-field-indicator"
            >
               *
            </span>
          </label>
          <select
            class="usa-select"
            data-testid="Select"
            id="inviteUser-role"
            name="role"
          >
            <option
              value=""
            >
              placeholder
            </option>
            <option
              value="1"
            >
              role_1
            </option>
          </select>
        </div>
      </div>
      <div
        class="tablet:grid-col-4 flex-align-self-end"
        data-testid="grid"
      >
        <button
          class="usa-button tablet:margin-top-0 margin-top-4"
          data-testid="button"
          type="submit"
        >
          <svg
            aria-hidden="true"
            class="usa-icon usa-icon margin-right-05 margin-left-neg-05"
            role="img"
          >
            <use
              href="/img.jpg#send"
            />
          </svg>
          label
        </button>
      </div>
    </div>
  </form>
</div>
`;
